src/PluginClient.php

```diff
@@ -79,7 +78,7 @@ public function sendRequest(RequestInterface $request)
         $pluginChain = $this->createPluginChain($this->plugins, function (RequestInterface $request) {
             try {
                 return new FulfilledPromise($this->client->sendRequest($request));
-            } catch (HttplugException $exception) {
+            } catch (\Exception $exception) {
```

Comment:
What about the RetryPlugin for example? It should not retry if it catches a non-httplug exception?

Also, If I would like to write my own History plugin to log all exceptions I cant do that?

The reson I do not think it is a BC break is because the return value of Plugin::handleRequest is a Promise. And a Promise can throw an \Exception. 

> So plugins need to expect (in typehints) that they might receive any kind of exceptions, but we only want to handle HTTPlug ones on purpose.

I assume by "we only want ..." you mean our plugins? Not all custom plugins as well?